smartfile/__init__.py

```diff
@@ -124,8 +124,40 @@ def put(self, endpoint, id=None, **kwargs):
     def post(self, endpoint, id=None, **kwargs):
         return self._request('post', endpoint, id=id, data=kwargs)
 
-    def delete(self, endpoint, id=None, **kwargs):
-        return self._request('delete', endpoint, id=id, data=kwargs)
+    def delete(self, deletefile):
+        try:
+            return self.post('/path/oper/remove', path=deletefile)
+        except KeyError:
+            raise Exception("Destination file does not exist")
+        
+    def upload(self, usrfile):
```

Comment:
I think a saner API would accept a filename (as it will represented on smartfile) and a fileobject, not a tuple.